### PR TITLE
statefulsets are updated during each sync even if no changes to the config

### PR DESCRIPTION
**Pull request by DanielZhangQD**
### What problem does this PR solve? 
statefulsets are updated during each sync even if no changes to the config
### What is changed and how does it work?
Fix the equality check logic
### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)
   - Check tidb-controller-manager logs that statefulsets are not updated without changes to the config

Code changes

 - Has Go code change

Related changes

 - Need to cherry-pick to the release branch

### Does this PR introduce a user-facing change?:

```release-note
Fix the issue that statefulsets are updated during each sync even if no changes to the config
```

### CODE REVIEW

**Comment by cofyc** [comment]
github actions failed, it's not possible to merge this PR. github actions must be re-run manually:

image

**Comment by DanielZhangQD** [comment]
> github actions failed, it's not possible to merge this PR. github actions jobs must be re-run manually:

OK, Thanks!

File: pkg/manager/member/utils.go [hunk 1]
```diff
@@ -100,7 +100,14 @@ func GetLastAppliedConfig(set *apps.StatefulSet) (*apps.StatefulSetSpec, *corev1
 
 // statefulSetEqual compares the new Statefulset's spec with old Statefulset's last applied config
 func statefulSetEqual(new apps.StatefulSet, old apps.StatefulSet) bool {
-	if !apiequality.Semantic.DeepEqual(new.Annotations, old.Annotations) {
+	// The annotations in old sts may include LastAppliedConfigAnnotation
+	var tmpAnno map[string]string
```
Comment (cofyc) [inline pkg/manager/member/utils.go:104]:
```suggestion
	tmpAnno := make(map[string]string)
```

File: pkg/manager/member/utils.go [hunk 2]
```diff
@@ -110,8 +117,16 @@ func statefulSetEqual(new apps.StatefulSet, old apps.StatefulSet) bool {
 			klog.Errorf("unmarshal Statefulset: [%s/%s]'s applied config failed,error: %v", old.GetNamespace(), old.GetName(), err)
 			return false
 		}
+		// oldConfig.Template.Annotations may include LastAppliedConfigAnnotation to keep backward compatiability
+		// Please check detail in https://github.com/pingcap/tidb-operator/pull/1489
+		tmpTemplate := oldConfig.Template.DeepCopy()
+		if tmpTemplate.Annotations != nil {
+			if _, ok := tmpTemplate.Annotations[LastAppliedConfigAnnotation]; ok {
+				delete(tmpTemplate.Annotations, LastAppliedConfigAnnotation)
+			}
```
Comment (cofyc) [inline pkg/manager/member/utils.go:126]:
```suggestion
			delete(tmpTemplate.Annotations, LastAppliedConfigAnnotation)
```

Comment (cofyc) [reply pkg/manager/member/utils.go:126]: more concise

File: pkg/manager/member/utils.go (hunk 1 above)
Comment (cofyc) [reply pkg/manager/member/utils.go:104]: fix panic

Comment (DanielZhangQD) [reply pkg/manager/member/utils.go]: Thanks a lot! Fixed.

File: pkg/manager/member/utils.go (hunk 2 above)
Comment (DanielZhangQD) [reply pkg/manager/member/utils.go]: Updated

**Pull request merged**